pravega/pravega
# Issue 4561: Using S3 Rest Style Config URI for Extended S3 client construction

Signed-off-by: kevinhan88 <kevinhan88@gmail.com>

**Change log description**  
Started using s3 rest style configUri for majority of extendeds3 storage client config.

**Purpose of the change**  
Closes #4561 

**What the code does**  
(Detailed description of the code changes)
config.properties now contains configUri, bucket and prefix for extendedS3 storage.
ExtendedS3StorageConfig takes configUri and construct s3Config from it.
ExtendedS3StorageFactory handles configUri now.
ExtendedS3StorageConfigTest is added.
ExtendedS3StorageTest is updated to work with the new configUri
ExtendedS3IntegrationTest is updated to work with the new configUri

**How to verify it**  
(Optional: steps to verify that the changes are effective)
No functional changes. All tests (unit, integration, system and longevity) should pass.
The new configUri should start taking effect.
